instrumentation/helpers-sql-obfuscation/lib/opentelemetry/instrumentation/helpers/sql_obfuscation.rb

```diff
@@ -0,0 +1,150 @@
+          Regexp.union(components.map { |component| COMPONENTS_REGEX_MAP[component] })
+        end
+
+        MYSQL_COMPONENTS_REGEX = generate_regex(:mysql)
+        POSTGRES_COMPONENTS_REGEX = generate_regex(:postgres)
+
+        # This is a SQL obfuscation utility intended for use in database adapter instrumentation.
+        #
+        # @param sql [String] The SQL to obfuscate.
+        # @param config [Hash] The user's configuration for the database adapter.
+        #   Required keys:
+        #     * `:db_statement` => A symbol with the values of either `:omit`, `:include`, or `:obfuscate`
+        #
+        #     * `:obfuscation_limit` => An integer representing the maximum length of a sql statement
+        #
+        #   Please include these keys in your instrumentation's configuration.
+        # @param adapter [Symbol] the type of database adapter calling the method. `:mysql` and `:postgres` are supported.
+        # @return [String] If successful, the SQL query string where the values are replaced with "?". If unsuccessful,
+        #  a string with a failure message.
+        #
+        # @api public
+        def self.obfuscate_sql(sql, config, adapter)
+          return sql unless config[:db_statement] == :obfuscate
+
+          regex = const_get("#{adapter.upcase}_COMPONENTS_REGEX")
+
+          return truncate_statement(sql, regex, config[:obfuscation_limit]) if sql.size > config[:obfuscation_limit]
+
+          obfuscated = utf8_encode_sql(sql, adapter).gsub(regex, PLACEHOLDER)
+          obfuscated = 'Failed to obfuscate SQL query - quote characters remained after obfuscation' if
+            detect_unmatched_pairs(obfuscated, adapter)
+          obfuscated
+        rescue StandardError => e
+          OpenTelemetry.handle_error(message: 'Failed to obfuscate SQL', exception: e)
+          'OpenTelemetry error: failed to obfuscate sql'
+        end
+
+        # @api private
+        def self.truncate_statement(sql, regex, limit)
+          # TODO: Make sure truncation is working
+          first_match_index = sql.index(regex)
+          truncation_message = "SQL truncated (> #{limit} characters)"
+          return truncation_message unless first_match_index
+
+          truncated_sql = sql[..first_match_index - 1]
+          "#{truncated_sql}...\n#{truncation_message}"
+        end
+
+        # @api private
+        def self.utf8_encode_sql(sql, adapter)
+          # postgres utf-8 encoding needs more work
+          return sql unless adapter == :mysql
+
+          # Original MySQL UTF-8 Encoding Fixes:
+          # https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/160
+          # https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/345
+          OpenTelemetry::Common::Utilities.utf8_encode(sql, binary: true)
+        end
+
+        # @api private
+        def self.detect_unmatched_pairs(obfuscated, adapter)
+          CLEANUP_REGEX[adapter].match(obfuscated)
```

Comment:
any decisions wrt `fallback` as a key on `DIALECT_COMPONENTS` should be mimic'd on `CLEANUP_REGEX`. Rn if an instrumentation passes in `fallback` as a value this will throw an error

Reply:
Fixed in: 8096551319f684cf850901dbc12100dee43c8694